x/twap/types/params.go

```diff
@@ -39,12 +48,30 @@ func (p Params) Validate() error {
 		return err
 	}
 
+	if err := validatePeriod(p.RecordHistoryKeepPeriod); err != nil {
+		return err
+	}
+
 	return nil
 }
 
 // Implements params.ParamSet.
 func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 	return paramtypes.ParamSetPairs{
 		paramtypes.NewParamSetPair(KeyPruneEpochIdentifier, &p.PruneEpochIdentifier, epochtypes.ValidateEpochIdentifierInterface),
+		paramtypes.NewParamSetPair(KeyRecordHistoryKeepPeriod, &p.RecordHistoryKeepPeriod, validatePeriod),
+	}
+}
+
+func validatePeriod(i interface{}) error {
```

Comment:
Can we please have a test for this function directly? We can use `export_test.go` to expose it

Reply:
I added testcase.